lib/matplotlib/artist.py

```diff
@@ -1380,6 +1380,27 @@ def _get_tightbbox_for_layout_only(obj, *args, **kwargs):
         return obj.get_tightbbox(*args, **kwargs)
 
 
+def getdoc_pure(object):
+    """
+    Get the documentation string for an object.
+
+    This is equal to `inspect.getdoc`, expect that the
+    function `inspect.cleandoc` is not applied.
+    """
+    try:
+        doc = object.__doc__
+    except AttributeError:
+        return None
+    if doc is None:
+        try:
+            doc = inspect._finddoc(object)
```

Comment:
As this is private, is it really something we can rely on across Python versions?

Reply:
@QuLogic Using the private method is indeed not really nice. It is something I expect to be stable (the code is stable https://github.com/python/cpython/blame/main/Lib/inspect.py#L786) and it is part of the unit testing, but no guarantees are given off course.

However, @timhoffm noted in https://github.com/matplotlib/matplotlib/pull/23686#issuecomment-1222732058 that the docstring parsing will be removed anyway, so I removed this part of the PR.